jetty-core/jetty-compression/jetty-compression-server/src/main/java/org/eclipse/jetty/compression/server/internal/DecompressionRequest.java

```diff
@@ -104,7 +105,7 @@ else if (field.containsLast(compression.getEncodingName()))
                 case IF_MATCH, IF_NONE_MATCH ->
                 {
                     String etags = field.getValue();
-                    String etagsNoSuffix = compression.stripSuffixes(etags);
+                    String etagsNoSuffix = EtagUtils.stripSuffixes(etags);
```

Comment:
The original one would only strip it's own suffix. (in case of nested handlers that manipulate the etag)
The ETagUtils strips all suffixes.

Reply:
exactly,  there is no link between the compression algorithm used to send this particular request and the one used to compress a previous response.... but I don't think I need this here now anyway... stand by....